### DISCUSSION TITLE
refer remote resources using caf super module

### DISCUSSION THREAD

**Discussion by hadiulla** [aztfmod/terraform-azurerm-caf#1775] [Q&A]
Hi guys
I have been using caf super module for quite sometime now and all works fine if I don't have to refer to any resources from other environments. I have two environments hub & spoke with their own state files. I find it very difficult to understand how to refer the the resource in spoke environment which was created in the hub. I know that caf supports that using lz_key but I haven't able to find a complete example which can explain this better.

I have create a sample repository with the current config. To keep it simple I am trying to refer to the resource group in spoke which was created in hub in automation resource. I am sure I am missing something like remote backend data source or something, but if someone can help me update the example, really appretiate it.
Here is the link to the repo - https://github.com/hadiulla/caf-example

Quick peek
This is what my automation account block looks like in spoke
```
automations = {
  account1 = {
    name = "automationAccount1"
    sku  = "Basic"
    resource_group = {
      key    = "rg1"
      lz_key = "networking_hub"
    }
  }
}
```

**Comment by khaman-pet** [comment]
This doc explains a bit about how to refer to other landingzone data. 
https://aztfmod.github.io/documentation/docs/fundamentals/lz-composition

**Reply by hadiulla** [reply]
The above document is not clear enough to understand on how to use this composition for standalone usage. I have read this document many times and tried things out, I am sure I am missing out something. Thats why I created a public repo as an example to fill the gap and can be of a help for other newbies like me.

**Reply by khaman-pet** [reply]
The lz_key can only be used if you are using the whole Landingzone framework for state management.

I believe this [local.remote_tfstates.tf](https://github.com/Azure/caf-terraform-landingzones/blob/37ccc595b1ee93c434ca547f93f39dfb9843b80f/caf_solution/local.remote_tfstates.tf) responsible on reading the data from other tfstate. 

PS. I'm also quite new to this framework. I'm using the whole framework and not as standalone module.
